WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -432,6 +458,23 @@ extension ProductFormViewModel {
         }
     }
 
+    func saveProductImagesWhenAllAreUploaded() {
+        productImagesUploader.saveProductImagesWhenNoneIsPendingUploadAnymore(siteID: product.siteID,
+                                                                              productID: product.productID,
+                                                                              isLocalID: !product.existsRemotely) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let images):
+                let currentProduct = self.product
+                self.originalProduct = .init(product: self.originalProduct.product.copy(images: images))
+                self.product = .init(product: currentProduct.product)
+            case .failure:
+                self.isUpdateEnabledSubject.send(self.hasUnsavedChanges())
```

Comment:
I hope we are doing this to show the save button as the product can be saved manually with the uploaded images. (As the update product request failed.) Please correct me if I am wrong.

Again, I think it will help to add a comment.

Reply:
Yes exactly, if the product images save request fails, the Save CTA should be shown again so that the merchant can save the product images again. Added a comment in https://github.com/woocommerce/woocommerce-ios/pull/7048/commits/9dd3efde0d2f70d710601cd8abf7ea90f30a8685